standalone-metastore/src/main/java/org/apache/hadoop/hive/metastore/txn/TxnHandler.java

```diff
@@ -1128,6 +1137,7 @@ public void commitTxn(CommitTxnRequest rqst)
   @Override
   @RetrySemantics.Idempotent("No-op if already replicated the writeid state")
   public void replTableWriteIdState(ReplTblWriteIdStateRequest rqst) throws MetaException {
+    String catName = rqst.isSetCatName() ? rqst.getCatName() : MetaStoreUtils.getDefaultCatalog(conf);
```

Comment:
This seems dangerous.  Could we require client to set catalog?  It would be safer to have as few places as possible calling getDefaultCatalog() - otherwise this may lead to hard to debug problems where some tables "jumps" from cat to cat.  There are a few more places where the same pattern is used

Reply:
I've built the catalog stuff to support several scenarios:
1. User explicitly sets a catalog.
2. Client is configured to use a particular catalog, usually a non-default one (e.g. Spark might use its own catalog rather than the Hive one).
3. Client is old and doesn't about the catalogs and thus leaves the value null.  
The checks here are to handle the last case.  I could move all of these checks into HiveMetaStore when it receives the calls and then have (Compaction)TxnHandler assume the catalog has been set.  That's mostly what I did on with ObjectStore.